teuthology/dispatcher/supervisor.py

```diff
@@ -151,6 +153,45 @@ def run_job(job_config, teuth_bin_path, archive_dir, verbose):
     if 'targets' in job_config:
         unlock_targets(job_config)
 
+def mark_machine_down(machine_name, count):
```

Comment:
I think the name of this function is too generic considering the implementation is very specific to reimage failures. I think it's okay to keep this inside the main function